### PR TITLE
nixos/swww: init at 0.7.3

### PR DESCRIPTION
**Pull request by b3nj5m1n**
###### Description of changes

This adds a systemd service for the [swww](https://github.com/Horus645/swww/) wallpaper daemon.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ocfox — COMMENTED** [review]
Is it possible to add an option to specify wallpaper?

**Comment by b3nj5m1n** [comment]
This module is for the swww daemon, while actually setting a wallpaper is handled by a client. (The client would be the `swww` binary, `swww init` actually calls the `swww-daemon` binary)

AFAIK, there's currently no option to specify an image on startup, and during all other calls you'll probably want to take advantage of swww's features, like transitions, which would add a lot of complexity to the module. (I've implemented a mechanism for changing wallpapers based on time of day in my personal config, so look [here](https://github.com/b3nj5m1n/dotfiles/blob/1da98eb2745ea4b0e4b789c38a037080bbf8347f/nix/modules/nixos/dynamic-wallpaper.nix) if you're interested)

I'd definitely be open to trying to implement some way of setting a default image on startup (`ExecStartPost` should work), but I think it would probably make more sense to implement that upstream instead of doing it through the systemd services.

Tldr: people who use swww probably set their wallpapers through scripts anyway; adding an option for setting the initial wallpaper may be a good idea, but it would probably be best to implement that as an option upstream.
I think our options are probably either merging this without such an option or implementing it in the systemd service and possibly changing it later, should swww get support for specifying this while starting the daemon.

**Comment by qweered** [comment]
any updates?
